### PR TITLE
Fix hyphen end of name in mail order confirmation

### LINKED ISSUE #35942: Hyphen after every product name in order_conf e-mail
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

In the order confirmation mail received by the customer, every product as an hyphen - added at the and of each product name.
image

### Expected behavior

Normally the hyphen is only added when product has a combination, like : Product name - Size - Color
Should be
image

### Steps to reproduce

1/ Add products to cart with and without combination
2/ Place order
3/ See confirmation order mail

### PrestaShop version(s) where the bug happened

1.7.8.11 to 8.1.5

### PHP version(s) where the bug happened

7.4

### If your bug is related to a module, specify its name and its version

_No response_

### Your company or customer's name goes here (if applicable).

@ComonSoft

### PR DESCRIPTION
**Pull request by ComonSoft**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch           | develop
| Description      | When a product doesn't have a combination an hyphen is added to the name in order_conf mail. Solution, test that attributes is not an empty string to add hyphen to product name.
| Type             | bug fix
| Category         | FO
| BC breaks        | no
| Deprecations     | no
| How to test      | Create an order with products with and without combination. The order confirmation mail contains hyphen at the end of product without combination
| UI Tests | https://github.com/florine2623/testing_pr/actions/runs/10111983173
| Fixed issue  | Fixes #35942.
| Sponsor company   | @ComonSoft

### CODE REVIEW

**Review by ComonSoft — COMMENTED** [review]
Last modification to be compliant to PHP / PHP CS Fixer

File: classes/PaymentModule.php [hunk 1]
```diff
@@ -457,7 +457,7 @@ public function validateOrder(
                     'id_product' => $product['id_product'],
                     'id_product_attribute' => $product['id_product_attribute'],
                     'reference' => $product['reference'],
-                    'name' => $product['name'] . (isset($product['attributes']) ? ' - ' . $product['attributes'] : ''),
+                    'name' => $product['name'] . (isset($product['attributes']) && $product['attributes'] != '' ? ' - ' . $product['attributes'] : ''),
```
Comment (kpodemski) [inline classes/PaymentModule.php]:
```suggestion
                    'name' => (!empty($product['attributes']) ? ' - ' . $product['attributes'] : ''),
```

maybe something simpler?

File: classes/PaymentModule.php [hunk 2]
```diff
@@ -457,7 +457,7 @@ public function validateOrder(
                     'id_product' => $product['id_product'],
                     'id_product_attribute' => $product['id_product_attribute'],
                     'reference' => $product['reference'],
-                    'name' => $product['name'] . (isset($product['attributes']) ? ' - ' . $product['attributes'] : ''),
+                    'name' => (!empty($product['attributes']) ? ' - ' . $product['attributes'] : ''),
```
Comment (Seb33300) [inline classes/PaymentModule.php]: Why did you remove `$product['name'] . `?

Comment (kpodemski) [reply classes/PaymentModule.php]: I think it's a mistake coming from my review

Comment (kpodemski) [reply classes/PaymentModule.php]: @Seb33300 I fixed it 👍🏻

**Review by Seb33300 — APPROVED** [review]

File: classes/PaymentModule.php (hunk 2 above)
Comment (ComonSoft) [reply classes/PaymentModule.php]: Yes there is a bug, because it has been removed when I applied the suggestion :-(

**Review by tleon — APPROVED** [review]

**Review by kpodemski — APPROVED** [review]

**Review by florine2623 — APPROVED** [review]
Hello @ComonSoft ,

Thanks for the PR ✅ 
Screenshot 2024-07-26 at 15 53 50

LGTM !

**Comment by kpodemski** [comment]
thank you @ComonSoft !

**Pull request merged**